File: b1616.py
MAX = 1_000
MOD = 1_000_000_007


def main() -> None:
    m = [[1 for j in range(MAX + 1)] for i in range(MAX + 1)]

    for i in range(1, MAX + 1):
        m[i][i] = (i * m[i - 1][i - 1]) % MOD

    for i in range(2, MAX + 1):
        for j in range(i + 1, MAX + 1):
            m[i][j] = (i * m[i - 1][j - 1] % MOD + i * m[i][j - 1] % MOD) % MOD

    b, g = map(int, input().split())
    while b and g:
        print(m[g][b])
        b, g = map(int, input().split())


if __name__ == '__main__':
    main()

File: b1616.cpp
// https://www.beecrowd.com.br/judge/pt/problems/view/1616

#include <bits/stdc++.h>

using namespace std;

#define _ std::ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
#define dbg(x) cout << #x << " = " << x << endl
#define endl "\n"
#define MAX 1000
#define MOD 1'000'000'007

typedef unsigned uint;
typedef unsigned long long ull;

int main() { _
    vector m(MAX + 1, vector<ull>(MAX + 1, 1));

    for (uint i {1}; i <= MAX; i++)
        m[i][i] = (i * m[i - 1][i - 1]) % MOD;

    for (uint i {2}; i <= MAX; i++)
        for (uint j {i + 1}; j <= MAX; j++)
            m[i][j] = (i * m[i - 1][j - 1] % MOD + i * m[i][j - 1] % MOD) % MOD;

    for (uint b, g; cin >> b >> g and b and g; )
        cout << m[g][b] << endl;
    
    return 0;
}
